tests/e2e/vsphere_volume_expansion.go

```diff
@@ -2504,6 +2511,154 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
 
 	})
 
+	/**
+	Offline and online volume expansion on stretched SVC
+	    1. Create zonal storage policy
+		2. Create PVC which uses the above sc.
+		3. Wait for PV to be provisioned.
+		4. Wait for PVC's status to become Bound and note down the size
+		5. Modify PVC's size to trigger offline volume expansion
+		6. verify the PVC status will change to "Resizing", but resize won't succeed until POD is created
+		9. Create POD
+		10. Verify PVC resize completes
+		11. Verify the resized PVC by doing CNS query
+		12. Modify PVC's size to trigger online volume expansion
+		12. Make sure data is intact on the PV mounted on the pod
+		13. After resize completes verify and Make sure file system has also increased
+		14. clean up the data
+	*/
+	ginkgo.It("[stretched-svc] offline and online volume "+
+		"expansion on stretched svc", ginkgo.Label(p1, block, newTest), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var pvclaims []*v1.PersistentVolumeClaim
+		//var zonalPolicy string
+		var categories []string
+		var allowedTopologyHAMap map[string][]string
+
+		scParameters := make(map[string]string)
+
+		topologyHaMap := GetAndExpectStringEnvVar(topologyHaMap)
+		_, categories = createTopologyMapLevel5(topologyHaMap)
+		allowedTopologies := createAllowedTopolgies(topologyHaMap)
+		allowedTopologyHAMap = createAllowedTopologiesMap(allowedTopologies)
+		framework.Logf("Topology map: %v, categories: %v", allowedTopologyHAMap, categories)
+
+		zonalPolicy := GetAndExpectStringEnvVar(envZonalStoragePolicyName)
+		scParameters[svStorageClassName] = zonalPolicy
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, zonalPolicy, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		pvclaim, err := createPVC(ctx, client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		pvclaims = append(pvclaims, pvclaim)
+		pv, err := fpv.WaitForPVClaimBoundPhase(ctx, client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volHandle := pv[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(ctx, client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Modify PVC spec to trigger volume expansion
+		// We expand the PVC while no pod is using it to ensure offline expansion
+		ginkgo.By("Expanding current pvc")
```

Comment:
Can we move these lines of expanding PVC size to a new util as it is used in many places

Reply:
I will take this in next PR. After changing this require to run on both single SVC, stretched SVC and vanilla